fetch/opensearch.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import absolute_import
+
+import logging
+import os
+
+from sentinelsat import SentinelAPI
+
+from ._core import DataSource, fetch_file
+
+_log = logging.getLogger(__name__)
+
+
+class OpenSearchApiSource(DataSource):
+    """
+    Class for data retrievals using the OpenSearch API.
+    """
+
+    def __init__(self, target_dir, api_url, username, password, query, show_progressbars=False, timeout=None,
```

Comment:
We should set a default timeout, otherwise a network hiccup can leave the spawned task running indefinitely.

See the HTTP sources for an example of a hard-coded default timeout: https://github.com/GeoscienceAustralia/fetch/blob/6d4b6f77f462f09f2a31c83a8c8c82437fec9684/fetch/http.py#L245

Reply:
(it's much more important on an ever-running service like fetch than on interactive scripts)